chasm/tree.go

```diff
@@ -650,13 +742,6 @@ func (n *Node) deserializeComponentNode(
 ) error {
 	// valueT is guaranteed to be a pointer to the struct because it was already validated by the assertStructPointer method.
 	valueV := reflect.New(valueT.Elem())
-	if n.serializedNode.GetData() == nil {
-		// serializedNode is empty (has only metadata) => use constructed value of valueT type as value and return.
-		// deserialize method acts as a component constructor.
-		n.value = valueV.Interface()
-		n.valueState = valueStateSynced
-		return nil
-	}
```

Comment:
This is sort of unrelated bug. We need to go through all sub fields even if `data` is `nil`. I discovered it with collection related tests.